### ISSUE TITLE
wav2vec CTC val_wer is always 1

### ISSUE THREAD

**Issue by demsarjure** [NVIDIA/NeMo#4506]
**Describe the bug**

Hi, I am having issues with training wav2vec models. It seems like the model starts training, there are no warnings or errors, everything runs. However, after every epoch val_wer is equal to 1.0. For [wav2vec_pretrain.yaml](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/ssl/wav2vec/wav2vec_pretrain.yaml) I am not sure where to start as there are no tutorials or examples. So I have been using [wav2vecCTC.yaml](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/wav2vec_ctc/wav2vecCTC.yaml) and the example script for training CTC models.

**Steps/Code to reproduce bug**

Below is the exact used command, used datasets work perfectly with other NeMo ASR models. It is the same command that I use for all CTC models.

```
python /workspace/nemo/examples/asr/asr_ctc/speech_to_text_ctc_bpe.py \\
  --config-path=/workspace/nemo/examples/asr/conf/wav2vec_ctc \\
  --config-name=wav2vecCTC \\
  exp_manager.name=wav2vec} \\
  +exp_manager.exp_dir=/experiments \\
  exp_manager.resume_if_exists=true \\
  exp_manager.resume_ignore_no_checkpoint=true \\
  +exp_manager.checkpoint_callback_params.save_best_model=true \\
  model.train_ds.manifest_filepath=/data/train.nemo \\
  model.train_ds.batch_size=8 \\
  model.train_ds.num_workers=4 \\
  model.train_ds.max_duration=17 \\
  +model.train_ds.min_duration=1 \\
  +model.train_ds.parser=base \\
  model.validation_ds.manifest_filepath=/data/validation.nemo \\
  model.validation_ds.batch_size=8 \\
  model.validation_ds.num_workers=4 \\
  +model.validation_ds.parser=base \\
  model.tokenizer.dir=/data/out/tokenizer_spe_unigram_v128 \\
  model.tokenizer.type=bpe \\
  trainer.precision=bf16 \\
  trainer.devices=-1
```

**Expected behavior**

I expect val_wer to drop with training and not remain at its maximum value (1.0).

**Environment overview (please complete the following information)**

 - Official latest NeMo container where the NeMo repository is updated to the current tip of the master branch.

**Environment details**

 - Official latest NeMo container where the NeMo repository is updated to the current tip of the master branch.

**Additional context**

Running on 4 RTX A6000 GPUs.

**Comment by demsarjure** [comment]
I will increase the `min_duration` and run again. I will let you know how it goes.

In first report at epoch 0 loss was above 800, after 15 epochs (I terminated training then) it dropped to 130. Like I said, val_wer for all epochs was 1.0.

**Comment by titu1994** [comment]
This is CTC finetuning right ? Not SSL. Are you attempting to train a Wav2Vec model from scratch without any SSL pretraining ? Dunno if we ever tried that. Should work, but it's hyper parameters will need to be tuned

**Comment by tbartley94** [comment]
@titu1994 We tried it earlier in the year but didn't get an ngc checkpoint up. It works and gets decent scoring iirc. I'm just trying to dig up the training logs to figure our the high validation WER because I remember it taking a big before it starts generalizing well. Probably something up with how the encoder passes encodings to the decoder.

**Comment by tbartley94** [comment]
@demarjure Just checked the logs back when we were training. It's not uncommon for the Validation WER to stall up at 1 for a good bit. (It took about 10 epochs for a model with 16 node training, for instance). Still doing some upkeep on it so can let you know if there's any issue, but it's not outside of reason for the model to take a good bit until the WER drops. For early training I'd focus on the validation loss and make sure that's not stalling.

**Comment by demsarjure** [comment]
Thanks, I will keep the training running a bit longer than.

If I wanted to perform SSL pre-training I need to use [wav2vec_pretrain.yaml](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/ssl/wav2vec/wav2vec_pretrain.yaml) right? Is there an example script for this or if you can point me in the right direction so I can maybe set this up first. Thanks!

**Comment by tbartley94** [comment]
You would use the same setup as any of the other SSL models from NeMo. You'll note in the yaml that the Wav2Vec specific portions are just two modules. So you can mess around with your own yaml file if you want. 

For pretraining can use this script: https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_pretraining/speech_pre_training.py 

And can doublecheck the SSL tutorial if any issues popup in getting setup. (Only major difference with standard models is Wav2Vec uses a different preprocessor.): 
https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Self_Supervised_Pre_Training.ipynb

You could also try simply carrying over the weights from the FAIR model like mentioned in https://github.com/NVIDIA/NeMo/issues/4497 . Some of the parameters need to be renamed (hence the aforementioned upkeep) but the mappings are near 1-to-1 at the moment, and tinkering with the namespace could take less time than pretraining.

**Comment by demsarjure** [comment]
Thanks for an extensive answer. I think I should be able to get things running now.
